### PR TITLE
cicleci: add ability use target branch for pulling focalboard

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
We are now going to pull target branch for focalboard. The only problem here is that the build started before opening the PR will checkout `rolling-stable` branch but once it's re-run it will get the target branch for the PR.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-49372

#### Release Note

```release-note
NONE
```

[MM-49372]: https://mattermost.atlassian.net/browse/MM-49372?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by agnivade** [comment]
> The only problem here is that the build started before opening the PR will checkout rolling-stable branch but once it's re-run it will get the target branch for the PR.

That's okay, we just need an `/update-branch` right? It's the same problem as sending PRs for server+enterprise both. The first one will always fail.

**Comment by isacikgoz** [comment]
> That's okay, we just need an `/update-branch` right? It's the same problem as sending PRs for server+enterprise both. The first one will always fail.

The `ee-build` runs after the PR is created so if you first push mattermost-server, then push enterprise and then create the PR, it wouldn't fail. But in this case it will always run before creating the PR. So we just need to re-run the build. Alternatively we can move build step after the PR is created.

File: .circleci/config.yml [hunk 1]
```diff
@@ -21,6 +21,10 @@ executors:
 
 jobs:
   setup-multi-product-repositories:
+    parameters:
+      target-branch:
+        type: string
+        default: '$(curl -s -H "Authorization: token ${GITHUB_TOKEN}" $(echo https://api.github.com/repos/${CIRCLE_PULL_REQUEST:19} | sed "s/\/pull\//\/pulls\//") | jq ".base.ref" | tr -d "\042" )'
```
Comment (agnivade) [inline .circleci/config.yml:27]: Can you clarify what 19 is? I am assuming it's a substring, just want to check. Also what is `\042`?

Comment (isacikgoz) [reply .circleci/config.yml:27]: Yeah it's the substring, we just remove the `https://github.com/` part from the URL. And `\042` is the quotes to be removed with `tr -d` from the branch name.

**Review by agnivade — APPROVED** [review]

**Pull request merged**